Packages/vcs/Lib/vcsvtk/boxfillpipeline.py

```diff
@@ -0,0 +1,383 @@
+    def _createPolyDataFilter(self):
+        """Overrides baseclass implementation."""
+        self._vtkPolyDataFilter = vtk.vtkDataSetSurfaceFilter()
+        if self._useCellScalars:
+            p2c = vtk.vtkPointDataToCellData()
+            p2c.SetInputData(self._vtkDataSet)
+            self._vtkPolyDataFilter.SetInputConnection(p2c.GetOutputPort())
+        else:
+            self._vtkPolyDataFilter.SetInputData(self._vtkDataSet)
+
+    def _plotInternal(self):
+        """Overrides baseclass implementation."""
+        # Special case for custom boxfills:
+        if self._gm.boxfill_type != "custom":
+            self._plotInternalBoxfill()
+        else:
+            self._plotInternalCustomBoxfill()
+
+        if self._maskedDataMapper is not None:
+            self._mappers.insert(0, self._maskedDataMapper)
+
+        x1, x2, y1, y2 = vcs.utils.getworldcoordinates(self._gm,
+                                                       self._data1.getAxis(-1),
+                                                       self._data1.getAxis(-2))
+
+        # And now we need actors to actually render this thing
+        actors = []
+        for mapper in self._mappers:
+            act = vtk.vtkActor()
+
+            # TODO Charles, why would mapper be a list? The custom boxfill code
+            # pushes single-element lists into the mapper list, though a comment
+            # suggests that this means a textured pattern should be used. I'll
+            # clean this up to be more clear, but need to understand the current
+            # usage pattern first.
+            if isinstance(mapper, list):
+                act.SetMapper(mapper[0])
+            else:
+                mapper.Update()
+                act.SetMapper(mapper)
+
+            if self._vtkGeoTransform is None:
+                # If using geofilter on wireframed does not get wrppaed not sure
+                # why so sticking to many mappers
+                act = vcs2vtk.doWrap(act, [x1,x2,y1,y2], self._dataWrapRange)
+
+            # TODO Charles, what's the difference between these two branches?
+            # Why not just store the actor and use GetMapper() to get the mapper
+            # when needed? The distinction isn't very clear to other developers
+            # that need to read the code.
+            if mapper is self._maskedDataMapper:
+                actors.append([act, self._maskedDataMapper, [x1,x2,y1,y2]])
+            else:
+                actors.append([act, [x1,x2,y1,y2]])
+
+            # TODO Charles, the custom boxfill code *will* trigger this block to
+            # execute, but without mapper[1] defined. Why are some entries in
+            # mappers actual mappers while others are single-element lists
+            # containing a mapper?
+            if isinstance(mapper, list):
+                ## This is the sport to add patterns
+                #act.GetMapper().ScalarVisibilityOff()
+                #act.SetTexture(mapper[1])
+                pass
```

Comment:
@doutriaux1 This code is commented out, but seems to be tied into the "mappers contains mappers and sometimes lists" issue. Note that none of the implementations would produce an entry in mappers that would have a second element (`mapper[1]` would throw a bounds error).